io/src/ply_io.cpp

```diff
@@ -543,6 +543,7 @@ pcl::PLYReader::parse (const std::string& istream_filename)
 
   pcl::io::ply::ply_parser::list_property_definition_callbacks_type list_property_definition_callbacks;
   pcl::io::ply::ply_parser::at<pcl::io::ply::uint8, pcl::io::ply::int32> (list_property_definition_callbacks) = [this] (const std::string& element_name, const std::string& property_name) { return listPropertyDefinitionCallback<pcl::io::ply::uint8, pcl::io::ply::int32> (element_name, property_name); };
+  pcl::io::ply::ply_parser::at<pcl::io::ply::uint8, pcl::io::ply::uint32> (list_property_definition_callbacks) = [this] (const std::string& element_name, const std::string& property_name) { return listPropertyDefinitionCallback<pcl::io::ply::uint8, pcl::io::ply::int32> (element_name, property_name); };
```

Comment:
Oof. I can see a  few combinations missing... but this changeset LGTM

Reply:
I am not that familiar with the PLY reader/parser, so just to be sure: the `int32` near the end is not supposed to be `uint32`?

Reply:
Yes, if  index will be more than `2^31`, than `int32` will be negative and `uint32` will not.
I think, this case are unlikely. The size of cloud must be more than 25GB. In addition if `pcl::index_t` == `int32`, so pcl can't read this cloud to `pcl::PolygonMesh` in any case.

Reply:
Its just that there is **uint32** in `at<pcl::io::ply::uint8, pcl::io::ply::uint32>` at the beginning and **int32** in `listPropertyDefinitionCallback<pcl::io::ply::uint8, pcl::io::ply::int32>` at the end of line.

Shouldn't they either be **int32**  or **uint32** at both places?

Reply:
As described in #4491, implementation for parsing `property list` exists only for pair `<uin8, int32>`

Function declaration looks like this:
```cpp
 /** \brief function called when a list property is parsed
   * \param[in] element_name element name to which the property belongs
   * \param[in] property_name list property name
   */
 template <typename SizeType, typename ScalarType>
 std::tuple<std::function<void (SizeType)>, std::function<void (ScalarType)>, std::function<void ()> >
 listPropertyDefinitionCallback (const std::string& element_name, const std::string& property_name);
```
I could replace implementation `<uin8, int32>` by specialization `<uint8, std::numeric_limits<T>::is_integer>`.
Specialization will be looks like this:
```cpp
template <typename SizeType, typename ScalarType>
typename std::enable_if<
    std::is_same<pcl::io::ply::uint8, SizeType>::value && std::numeric_limits<ScalarType>::is_integer,
    std::tuple<std::function<void (SizeType)>, std::function<void (ScalarType)>, std::function<void ()> >
>::type
listPropertyDefinitionCallback (const std::string& element_name, const std::string& property_name);
```
Is this will be better?

Reply:
I'm just puzzled why its not unit32 both places - but lets just keep your current implementation.